keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -243,147 +242,161 @@ A `Bucket` is not deleted if it is bound to a `BucketRequest`.
 
 #### BucketClass
 
-An immutable, cluster-scoped, custom resource to provide admins control over the handling of bucket provisioning.  The `BucketClass` (BC) defines a retention policy, driver specific parameters, and the provisioner name. A list of allowed namespaces can be specified to restrict new bucket creation and access to existing buckets. A default bucket class can be defined for each supported protocol. This allows the bucket class to be omitted from a `BucketRequest`. Relevant `BucketClass` fields are copied to the `Bucket` instance to handle the case of the BC being deleted or re-created.  If an object store supports more than one protocol then the admin should create a `BucketClass` per protocol.
+An immutable, cluster-scoped, greenfield-only, custom resource to provide admins control over the handling of bucket provisioning. The `BucketClass` (BC) is referenced by a [BucketRequest](#bucketrequest) (BR) for the purpose of instantiating a [Bucket](#bucket) and creating a new backend bucket. A BC defines a deletion policy, driver specific parameters, and the provisioner name. A list of allowed namespaces can be specified to restrict bucket creation to specific namespaces. A default bucket class can be defined which allows the bucket class to be omitted from a BR. Relevant BC fields are copied to the `Bucket` instance to handle the case of the BC being deleted and re-created.  If an object store supports more than one protocol then the admin should create a `BucketClass` per protocol.
 
-NOTE: the `BucketClass` object is immutable except for the field `isDefaultBucketClass`
+A `BucketClass` is required for greenfield use cases. It is not used for brownfield since the information in the BC will be defined in the `Bucket` instance.
+
+> Note: the `BucketClass` object is immutable except for the `isDefault` annotation.
 
 ```yaml
-apiVersion: cosi.io/v1alpha1
+apiVersion: objectstorage.k8s.io/v1alpha1
 kind: BucketClass
 metadata:
   name: 
-provisioner: [1]
-isDefaultBucketClass: [2]
-protocol:
-  name: [3]
-  version: [4]
-anonymousAccessMode: [5]
-retentionPolicy: {"Delete", "Retain"} [6]
-allowedNamespaces: [7]
+  annotations:
+    objectstorage.k8s.io/is-default-class: [1]
+provisioner: [2]
+protocol: [3]
+  azureBlob:
```

Comment:
We capture protocol rather than vendor drivers. The equivalent to this in block and file storage would be capturing POSIX for FS, and SCSI/ATA/NVMe for block device. This is a small list, and we don't expect this to grow

Reply:
Do none of the other clouds have their own protocols or dialects of protocols ?  The usual pattern here would be an object-ref to an S3-specific type.  I'm not going to fight too hard on this one, but it does have a familar smell.